From 15e3ba0f7bbf0adec8ca93d6acaf5f488582c949 Mon Sep 17 00:00:00 2001
From: Gary Gregory <gardgregory@gmail.com>
Date: Thu, 28 Feb 2019 09:12:26 -0500
Subject: [PATCH] [DBCP-538] Wrong JMX base name derived in
 BasicDataSource#updateJmxName.

- Add public org.apache.commons.dbcp2.ObjectNameWrapper.toString().
- Next version will be 2.7.0 (new public method, see above.)
---
 pom.xml                                                  | 4 ++--
 src/changes/changes.xml                                  | 3 +++
 .../java/org/apache/commons/dbcp2/ObjectNameWrapper.java | 9 +++++++++
 3 files changed, 14 insertions(+), 2 deletions(-)

diff --git a/pom.xml b/pom.xml
index dcfe31f2a2..a7b15a37fd 100644
--- a/pom.xml
+++ b/pom.xml
@@ -26,7 +26,7 @@
   </parent>
   <modelVersion>4.0.0</modelVersion>
   <artifactId>commons-dbcp2</artifactId>
-  <version>2.6.1-SNAPSHOT</version>
+  <version>2.7.0-SNAPSHOT</version>
   <name>Apache Commons DBCP</name>
 
   <inceptionYear>2001</inceptionYear>
@@ -286,7 +286,7 @@
     <commons.rc.version>RC1</commons.rc.version>
     <commons.module.name>org.apache.commons.dbcp2</commons.module.name>
     
-    <commons.release.version>2.6.1</commons.release.version>
+    <commons.release.version>2.7.0</commons.release.version>
     <commons.release.desc>for JDBC 4.2 on Java 8</commons.release.desc>
     
     <commons.release.2.version>2.4.0</commons.release.2.version>
diff --git a/src/changes/changes.xml b/src/changes/changes.xml
index 7bd5b5c191..c00b2c104d 100644
--- a/src/changes/changes.xml
+++ b/src/changes/changes.xml
@@ -64,6 +64,9 @@ The <action> type attribute can be add,update,fix,remove.
       <action dev="jleroux" type="add" issue="DBCP-539" due-to="Jacques Le Roux">
         ManagedDataSource::close should declare used exceptions.
       </action>
+      <action dev="ggregory" type="add" issue="DBCP-538" due-to="Ragnar Haugan, Gary Gregory">
+        Wrong JMX base name derived in BasicDataSource#updateJmxName.
+      </action>
     </release>
     <release version="2.6.0" date="2019-02-14" description="This is a minor release, including bug fixes and enhancements.">
       <action dev="chtompki" type="add" issue="DBCP-534" due-to="Peter Wicks">
diff --git a/src/main/java/org/apache/commons/dbcp2/ObjectNameWrapper.java b/src/main/java/org/apache/commons/dbcp2/ObjectNameWrapper.java
index baf983d90d..1cecc98dcb 100644
--- a/src/main/java/org/apache/commons/dbcp2/ObjectNameWrapper.java
+++ b/src/main/java/org/apache/commons/dbcp2/ObjectNameWrapper.java
@@ -18,6 +18,7 @@
 package org.apache.commons.dbcp2;
 
 import java.lang.management.ManagementFactory;
+import java.util.Objects;
 
 import javax.management.MBeanServer;
 import javax.management.MalformedObjectNameException;
@@ -76,6 +77,14 @@ public void registerMBean(final Object object) {
         }
     }
 
+    /**
+     * @since 2.7.0
+     */
+    @Override
+    public String toString() {
+        return Objects.toString(objectName);
+    }
+
     public void unregisterMBean() {
         if (MBEAN_SERVER == null || objectName == null) {
             return;
